### ISSUE TITLE
Consistent directory structure

### ISSUE THREAD

**Issue by langston-barrett** [GaloisInc/macaw#427]
Most of the packages in this repo live in directories that match their package names, e.g., `macaw-ppc`. However, some do not:

- [ ] `base`: `macaw-base`
- [ ] `refinement`: `macaw-refinement`
- [ ] `symbolic`: `macaw-symbolic`
- [ ] `symbolic-syntax`: `macaw-symbolic-syntax`
- [ ] `x86`: `macaw-x86`
- [ ] `x86_symbolic`: `macaw-x86-symbolic`

For the sake of consistency, I propose we rename all of the top-level directories according to a consistent naming scheme. I personally have no preference about whether this should include the `macaw-` prefix or not. At a bare minimum, I think it should exclusively use hyphens `-` or underscores `_` between words, and should probably use hyphens (to match the Cabal package convention, and the majority of existing directories in the repo).

**Comment by RyanGlScott** [comment]
I agree that we should consistently use a `macaw-` prefix. We could even add a [`.git-blame-ignore-revs` file](https://gist.github.com/kateinoigakukun/b0bc920e587851bfffa98b9e279175f2) to teach `git blame` to ignore the commit that moves the directories from the old names to the new ones.

**Comment by kquick** [comment]
This is likely going to create some churn in downstream users of macaw (updating cabal.project, etc.), and some of those are not public projects and therefore won't be something addressable by the person making this change to macaw, so if it's just for aesthetic purposes I'm not sure it's worth it.

**Comment by RyanGlScott** [comment]
Indeed, some amount of downstream churn will be inevitable. That being said, I do think it would be worth doing this, as the amount of cognitive overhead I have to deal with in remembering which subdirectories begin with `macaw-*` and which do not is non-trivial.

Perhaps it would be worth doing this and the `macaw-loader` merger (proposed in https://github.com/GaloisInc/macaw/issues/283) at the same time so that we can get all of these naming changes out of the way at once.
